src/pray.c

```diff
@@ -3946,66 +3946,235 @@ aligntyp alignment;
 {
 	register struct obj *otmp;
 	const char *what = (const char *)0;
+	int i, ii, lim, timeout;
 	
 	if (rnl((30 + u.ulevel)*10) < 10) god_gives_pet(align_gname_full(alignment),alignment);
 	else {
-		switch (rnl(5)) {
-			case 0: /* randomly charge an object */
-			case 1: /* increase weapon bonus */
-				if(uwep && uwep->spe < 7 && (uwep->oclass == WEAPON_CLASS || is_weptool(uwep))){
+		switch (rn2(6)) {
+			case 0: // randomly increment an ability score
+				i = rn2(A_MAX);
+				for (ii = 0; ii < A_MAX; ii++) {
+					lim = AMAX(i);
+					if (i == A_STR && u.uhs >= 3) --lim;
+					if (ABASE(i) < lim) {
+						ABASE(i) = lim;
+						pline("Wow! You feel good!");
+						break;
+					}
+					if (++i >= A_MAX) i = 0;
+				}
+				
+				i = rn2(A_MAX);
+				for (ii = A_MAX; ii > 0; ii--) {
+					if (adjattrib(i, 1, (ii == 1) ? 0 : -1))
+						break;
+					i = (i+1)%6;
+				}
+				flags.botl = 1;
+				break;
+			case 1: // increase weapon enchantment
+				if (uwep && uwep->spe < 7 && (uwep->oclass == WEAPON_CLASS || is_weptool(uwep)))
 					uwep->spe++;
+				else if(uswapwep && uswapwep->spe < 7 && (uswapwep->oclass == WEAPON_CLASS || is_weptool(uswapwep)))
+					uswapwep->spe++;
+				
+				break;
+			case 2: // identify an item
+				if (uwep && not_fully_identified(uwep)) identify(uwep);
+				else if (uswapwep && not_fully_identified(uswapwep)) identify(uswapwep);
+				else if (uamul && not_fully_identified(uamul)) identify(uamul);
+				else if (uleft && not_fully_identified(uleft)) identify(uleft);
+				else if (uright && not_fully_identified(uright)) identify(uright);
+				else if (uarmc && not_fully_identified(uarmc)) identify(uarmc);
+				else if (uarm && not_fully_identified(uarm)) identify(uarm);
+				else if (uarmu && not_fully_identified(uarmu)) identify(uarmu);
+				else if (uarmh && not_fully_identified(uarmh)) identify(uarmh);
+				else if (uarmg && not_fully_identified(uarmg)) identify(uarmf);
+				else if (uarmf && not_fully_identified(uarmf)) identify(uarmf);
+				else if (uarms && not_fully_identified(uarms)) identify(uarms);
+				else {
+					for(otmp=invent; otmp; otmp=otmp->nobj)
+					if (not_fully_identified(otmp)) identify(otmp);
```

Comment:
Ahem, this identifies the rest of your inventory, not a single item.
```suggestion
					if (not_fully_identified(otmp)) {
						identify(otmp);
						break;
					}
```